### PR TITLE
TST: Fix dtype compatibility in io.fits.utils with numpy-dev

### PR DESCRIPTION
**Pull request by pllim**
Example log: https://travis-ci.org/astropy/astropy/jobs/538455547 

The numpy-dev job caught one more warning than expected:
```
_________________ TestSingleTable.test_simple_meta_conflicting _________________
self = <astropy.io.fits.tests.test_connect.TestSingleTable object at 0x7f9b100eacc0>
tmpdir = local('/tmp/pytest-of-travis/pytest-0/test_simple_meta_conflicting0')
    def test_simple_meta_conflicting(self, tmpdir):
        filename = str(tmpdir.join('test_simple.fits'))
        t1 = Table(self.data)
        t1.meta['ttype1'] = 'spam'
        with catch_warnings() as l:
            t1.write(filename, overwrite=True)
>       assert len(l) == 1
E       assert 2 == 1
E        +  where 2 = len([<warnings.WarningMessage object at 0x7f9b0fe06550>, <warnings.WarningMessage object at 0x7f9b0fe06c88>])
astropy/io/fits/tests/test_connect.py:90: AssertionError
```

This PR tried to get more information on that extra warning. And also it disables installing matplotlib dev for the same job due to matplotlib/matplotlib#14370 .

### CODE REVIEW

**Comment by pllim** [comment]
@mhvk , any advice on how to fix this failure?

```
E           Passing (type, 1) or '1type' as a synonym of type is deprecated;
in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
from .../astropy/io/fits/util.py
```

**Comment by mhvk** [comment]
That's on l768, right? I think one can just replace `(bytes, 1) ` by `'S1` - that should work for all versions of numpy.

**Comment by pllim** [comment]
@mhvk , there is another one here. Any idea? Thanks!

https://github.com/astropy/astropy/blob/62f1ac1726e57facf922eab9fa3ee36b03aa8a2d/astropy/io/fits/util.py#L905

**Comment by mhvk** [comment]
I think making item 0 look like a tuple should do it, i.e., `"({0},){1}u{2}".format(dt.itemsize // bpc, dt.byteorder, bpc)`

File: astropy/io/fits/tests/test_connect.py [hunk 1]
```diff
@@ -85,11 +85,16 @@ def test_simple_meta_conflicting(self, tmpdir):
...
         t1.meta['ttype1'] = 'spam'
-        with catch_warnings() as l:
+        with catch_warnings() as tstwarns:
             t1.write(filename, overwrite=True)
-        assert len(l) == 1
-        assert str(l[0].message).startswith(
-            'Meta-data keyword ttype1 will be ignored since it conflicts with a FITS reserved keyword')
+        if len(tstwarns) != 1:
+            errmsg = os.linesep.join(
+                ['{} from {}:L{}'.format(w.message, w.filename, w.lineno)
```
Comment (pllim) [inline astropy/io/fits/tests/test_connect.py]: When it is time to clean this up, should we keep this? Or revert all changes to this test? It is nice to keep for future debugging but maybe there won't be necessary future debugging here?

Comment (mhvk) [reply astropy/io/fits/tests/test_connect.py]: I think this should be removed - we have tests to check things work, and need to inspect interactively anyway if they fail.

**Comment by pllim** [comment]
Numpy-dev job is green again! Please approve if you are okay with this. Thanks!

**Pull request merged**